resolwe/flow/models/utils/duplicate.py

```diff
@@ -0,0 +1,301 @@
+"""Resolwe models duplicate utils."""
+from copy import deepcopy
+
+from django.contrib.auth.models import Permission
+from django.contrib.contenttypes.models import ContentType
+from django.core.exceptions import ValidationError
+from django.db import IntegrityError, models, transaction
+from django.utils import timezone
+from guardian.models import GroupObjectPermission, UserObjectPermission
+
+from resolwe.flow.utils import iterate_fields
+
+
+def _check_permissions(objects, permission, user):
+    """"""
+    for obj in objects:
+        if not user.has_perm(permission, obj):
+            raise ValidationError(
+                "User doesn't have '{}' permission on {}.".format(permission, obj)
+            )
+
+
+def _bulk_copy_permissions(content_type, pk_mapping):
+    for permission_cls in [GroupObjectPermission, UserObjectPermission]:
+
+        permissions_qs = permission_cls.objects.filter(
+            content_type=content_type, object_pk__in=pk_mapping.keys()
+        )
+        new_permissions = []
+
+        for permission in permissions_qs:
+            permission.pk = None
+            permission.object_pk = pk_mapping[int(permission.object_pk)]
```

Comment:
I believe this should be converted to string? Or is pk_mapping a map between int and string?

Reply:
This is automatically handled by Django. But you are right that it is nicer to explicitly cast it to a string.